dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -1807,6 +1781,49 @@ private boolean deleteContentlets(List<Contentlet> contentlets, User user,
         return noErrors;
     }
 
+    /**
+     * Verifies if a page is being used as a detail page for any content type
+     * @param user
+     * @param c
+     * @throws DotDataException
+     * @throws LanguageException
+     */
+    private void validateRelatedContentType(User user, Contentlet c)
+            throws DotDataException{
+        ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(user);
+        List<ContentType> relatedContentTypes = contentTypeAPI.search("page_detail='" + c.getIdentifier() + "'");
+        HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+        String uri = htmlPageAssetAPI.fromContentlet(c).getURI();
+        //Verifies if the page is related to any content type
+        if (relatedContentTypes != null && !relatedContentTypes.isEmpty()){
```

Comment:
There's a UtilMethods.isSet for collections than can be used for this two checks FWIW.